src/subcommand/merge_subcommand.cpp

```diff
@@ -85,23 +108,167 @@ void merge_subcommand::create_merge_commit(
 
     // TODO: add a prompt to edit the merge message
     std::string msg_target = merge_ref ? merge_ref->short_name() : git_oid_tostr_s(&(merge_commit.oid()));
+    msg_target = "\'" + msg_target + "\'";
 	std::string msg = merge_ref ? "Merge branch " : "Merge commit ";
-	msg.append(msg_target);
+	msg.append( msg_target);
 
 	repo.create_commit(author_committer_sign_now, msg, std::optional<commit_list_wrapper>(std::move(parents)));
 
 	repo.state_cleanup();
 }
 
+int populate_list(const git_oid* oid, void* payload)
```

Comment:
I think this one deserves a comment stating that this function is used as a callback in `git_repository_mergehead_foreach` and therefore its type must be `git_repository_mergehead_foreach_cb`.